### PR TITLE
Brendan/algorithm dev

### PR DESCRIPTION
**Pull request by Brendanlai**
Updated the algorithm and implemented the add to Spotify playlist endpoint

### CODE REVIEW

File: backend/Controllers/algorithmController.js [hunk 1]
```diff
@@ -14,9 +14,9 @@ const redirect_uri = backend_url + "callback"; // Your redirect uri
 
 // instantiate spotifyApi object
 var spotifyApi = new SpotifyWebApi({
-    clientId: client_id,
-    clientSecret: client_secret,
-    redirectUri: redirect_uri,
+  clientId: client_id,
+  clientSecret: client_secret,
+  redirectUri: redirect_uri,
 });
```
Comment (Remyzh) [inline backend/Controllers/algorithmController.js:20]: Is this the part where we connect to spotify API?

File: backend/Controllers/algorithmController.js [hunk 2]
```diff
@@ -26,198 +26,273 @@ var spotifyApi = new SpotifyWebApi({
...
-    // put all the songs into one array, the master set
-    for (x of usersTopTracks) {
-        for (data of x) {
-            masterSet.push(data);
-            masterIDs.push(data.trackID);
-        }
+    // Add each persons dataset to our master array
+    for (x of data) {
+      usersTopTracks.push(x.topTracks);
+      usersMusicalProfile.push(x.musicalProfile);
```
Comment (Remyzh) [inline backend/Controllers/algorithmController.js:46]: I'm a little confused, does x.topTracks return toptracks from spotify api and then pushes it to usersTopTracks? Same thing with musicalProfile?

**Review by KevinetPeng — APPROVED** [review]
Looks good to me!

**Review by Remyzh — APPROVED** [review]

File: backend/Controllers/algorithmController.js (hunk 1 above)
Comment (brendanlai) [reply backend/Controllers/algorithmController.js:20]: yes that instantiates the spotifAPI object

File: backend/Controllers/algorithmController.js (hunk 2 above)
Comment (brendanlai) [reply backend/Controllers/algorithmController.js:46]: In this endpoint there is no connection to SpotifyAPI and everything is taken from mongoDB. Here the data extracted originally from MongoDB is in a single array with each users top tracks as an item in that array. So this part is extracting each subitem within the user top tracks so we can add just the tracks into a single array

**Pull request merged**